cellxgene_schema_cli/tests/test_validate.py

```diff
@@ -355,20 +356,61 @@ def test_determine_seurat_convertibility(self):
 
 
 class TestValidatorValidateDataFrame:
-    def test_fail_category_not_string(self, tmp_path):
+    def test_succeed_categorical_str(self, tmp_path):
+        self._succeed_categorical(tmp_path, ["hello", "world"])
+
+    @pytest.mark.parametrize("typ", [numpy.int64, numpy.int32, int])
+    def test_succeed_categorical_int(self, tmp_path, typ):
+        self._succeed_categorical(tmp_path, [typ(123), typ(456)])
+
+    def test_succeed_mixed_categorical_int(self, tmp_path):
+        self._succeed_categorical(tmp_path, [numpy.int32(123), int(456)])
+
+    def test_fail_categorical_bool(self, tmp_path):
+        validator = self._fail_categorical(tmp_path, [True, False])
+        assert "Categories must only contain" in validator.errors[0]
+
+    @pytest.mark.parametrize("typ", [numpy.float64, numpy.float32, float])
+    def test_fail_categorical_float64(self, tmp_path, typ):
+        validator = self._fail_categorical(tmp_path, [typ(123.123), type(456.456)])
```

Comment:
I can't recreate the write labels error with float64. Can you share your snippet you use to test?

I used (on an existing h5ad):
```
categories = [np.float64(4), np.float64(5)]
t = pd.CategoricalDtype(categories=categories)
adata.obs["float_cat"] = np.random.choice(categories, adata.n_obs)
adata.obs["float_cat"] = adata.obs["float_cat"].astype("category")
```

And was able to write successfully

Reply:
I was also able to download the h5ads in prod that reported using floats and I was able to read and write on them

Reply:
I tried your way and it does work. When I use the methods:

```
        categories = [np.float32(4), np.float32(5)]
        t = pd.CategoricalDtype(categories=categories)
        adata.obs["not_string"] = pd.Series(data=categories, index=["X", "Y"], dtype=t)
```

I end up with two class types in the category. A `float` class type and `type` class type. I have no idea where the `type` class type came from. None of the other tests cases do this and the other tests are using the same method.

OMG I found it, it's a typo on my end. >> `type(456.456)`->`typ(456.456)``